spec/models/invoice_spec.rb

```diff
@@ -2,24 +2,121 @@
 
 RSpec.describe Invoice, :type => :model do
 
-  it "has no invoice number if it is a draft" do
-    invoice = FactoryGirl.create(:invoice, draft: true)
-    expect(invoice.number).to be_nil
+  def build_invoice(**kwargs)
+    kwargs[:series] = Series.new(value: "A") unless kwargs.has_key? :series
+    kwargs[:issue_date] = Date.current() unless kwargs.has_key? :issue_date
+
+    invoice = Invoice.new(**kwargs)
+    invoice.set_amounts
+    invoice
+  end
+
+  #
+  # Invoice Number
+  #
+
+  it "no need for an invoice number if it's a draft" do
+    invoice = build_invoice(draft: true)
+    invoice.save
+
+    expect(invoice.number).to be nil
+  end
+
+  it "gets an invoice number after saving if it's not a draft" do
+    invoice = build_invoice()
+    invoice.save
+
+    expect(invoice.number).to eq 1
+  end
+
+  it "may have the same number as another invoice from a different series" do
+    invoice1 = build_invoice(series: Series.new(value: "A"))
+    invoice1.save
+
+    invoice2 = build_invoice(series: Series.new(value: "B"))
+    invoice2.save
+
+    expect(invoice1.number).to eq 1
+    expect(invoice2.number).to eq invoice1.number
+  end
+
+  #
+  # Status
+  #
+
+  it "returns the right status: draft" do
+    invoice = build_invoice(draft: true)
+    expect(invoice.get_status()).to eq :draft
   end
 
-  it "has an invoice number if it is not a draft" do
-    invoice1 = FactoryGirl.create(:invoice)
-    expect(invoice1.number).to eq(1)
+  it "returns the right status: failed" do
+    invoice = build_invoice(failed: true)
+    expect(invoice.get_status()).to eq :failed
+  end
+
+  it "returns the right status: pending (1/2)" do
+    invoice = build_invoice(items: [Item.new(quantity: 1, unitary_cost: 10)])
+    expect(invoice.get_status()).to eq :pending
+    invoice.due_date = Date.current() + 1
```

Comment:
I forgot to remove this after splitting the test in two different cases. I'll do right now.